bentoml/deployment/sagemaker/__init__.py

```diff
@@ -343,27 +358,216 @@ def delete(self):
+            # We will also try to delete both model and endpoint configuration for user.
+            # Since they are not critical, even they failed, we will still count delete
+            # deployment a success action
+            model_name = create_sagemaker_model_name(
+                deployment_spec.bento_name, deployment_spec.bento_version
+            )
+            delete_model_response = sagemaker_client.delete_model(ModelName=model_name)
+            logger.info("AWS delete model response: %s", delete_model_response)
+            if delete_model_response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+                logger.error(
+                    "Encounter error when deleting model: %s", delete_model_response
+                )
+
+            endpoint_config_name = create_sagemaker_endpoint_config_name(
+                deployment_spec.bento_name, deployment_spec.bento_version
+            )
+            delete_endpoint_config_response = sagemaker_client.delete_endpoint_config(  # noqa: E501
+                EndpointConfigName=endpoint_config_name
+            )
+            logger.info(
+                "AWS delete endpoint config response: %s",
+                delete_endpoint_config_response,
+            )
+            return DeleteDeploymentResponse(status=Status.OK())
+        else:
+            raise BentoMLDeploymentException('Delete Sagemaker deployment failed')
+
+    def describe(self, deployment_pb, repo=None):
+        deployment_spec = deployment_pb.spec
+        sagemaker_config = deployment_spec.sagemaker_operator_config
+        if sagemaker_config is None:
+            raise BentoMLDeploymentException('Sagemaker configuration is missing.')
+        sagemaker_client = boto3.client('sagemaker', sagemaker_config.region)
+        endpoint_name = generate_aws_compatible_string(
+            deployment_pb.namespace + '-' + deployment_spec.bento_name
+        )
+        endpoint_status_response = sagemaker_client.describe_endpoint(
+            EndpointName=endpoint_name
+        )
+        logger.info("AWS describe endpoint response: %s", endpoint_status_response)
+        endpoint_status = endpoint_status_response["EndpointStatus"]
+
+        # Sagemaker response status: 'OutOfService'|'Creating'|'Updating'|
+        #                            'SystemUpdating'|'RollingBack'|'InService'|
+        #                            'Deleting'|'Failed'
+        if endpoint_status == 'InService':
+            service_state = DeploymentState.RUNNING
+        elif endpoint_status in [
+            'Creating',
+            'Updating',
+            'RollingBack',
+            'SystemUpdating',
+        ]:
+            service_state = DeploymentState.PENDING
+        elif endpoint_status == 'Deleting':
+            service_state = DeploymentState.INACTIVATED
+        elif endpoint_status == 'Failed':
+            service_state = DeploymentState.ERROR
```

Comment:
Should we surface errors to users for failed or undocumented status? Also feel using a dict might be better than many elif.

Reply:
good suggestion on dict instead of if/elif. 
I think we could just let the undocumented status fail, it will be caught at the outer level and respond correctly.  Since we dump everything into info_json, we can properly process and surface that in the next iteration.